### PR TITLE
GCW-3443 rake task to clear and merge locations

### PR DESCRIPTION
**Pull request by swatijadhav**
### Ticket Link: 
https://jira.crossroads.org.hk/browse/GCW-3443

### What does this PR do?
rake task to clear and merge locations

### CODE REVIEW

File: lib/tasks/locations.rake [hunk 1]
```diff
@@ -0,0 +1,53 @@
+namespace :goodcity do
+
+  # rails goodcity:remove_unused_locations
+  desc 'Delete locations which does not have any associated data'
+  task remove_unused_locations: :environment do
+    Location.find_in_batches(batch_size: 100).each do |locations|
```
Comment (bharat619) [inline lib/tasks/locations.rake:6]: should we move the task to a new / existing lib file
and support the operations with specs?

File: lib/tasks/locations.rake [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          puts "Deleted location: #{location.id} #{location.building}-#{location.area}"
+        end
+      end
+    end
+  end
+
+  # rails goodcity:merge_locations
+  desc 'Move data from one location to another and delete location'
+  task merge_locations: :environment do
+    source_location_id = ENV['SOURCE_LOCATION_ID']
```
Comment (bharat619) [inline lib/tasks/locations.rake:19]: same as above

File: lib/tasks/locations.rake (hunk 1 above)
Comment (patrixr) [reply lib/tasks/locations.rake:6]: I think we almost always want to enrich the capabilities of goodcity with tested operations, and call them from rakes

**Review by abulsayyad123 — APPROVED** [review]

File: lib/classes/manage_location.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class ManageLocation
+  def initialize(location_id)
+    @location = Location.find_by(id: location_id)
+  end
+
+  def empty_location?
+    @location.packages_locations.count.zero? &&
+      @location.package_types.count.zero? &&
+      PackagesInventory.where(location_id: @location.id).count.zero? &&
```
Comment (patrixr) [inline lib/classes/manage_location.rb:9]: There will likely be cases where the we have packages_inventories records, but the quantity is 0. Those wouldn't be detected with this, right ?

**Review by patrixr — APPROVED** [review]
Looks pretty safe, although I dropped in a question : )

File: lib/classes/manage_location.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      Printer.where(location_id: @location.id).count.zero? &&
+      Stocktake.where(location_id: @location.id).count.zero?
+  end
+
+  def self.merge_location(source_location, destination_location)
+    %w[Printer Stocktake PackagesInventory PackageType PackagesLocation Package].each do |klass|
```
Comment (patrixr) [inline lib/classes/manage_location.rb:15]: Do packages have a `location_id` ? If so we should probably remove that column

Comment (swatijadhav) [reply lib/classes/manage_location.rb:15]: Yeah it is there, maybe an old column as it is not used anywhere. But just to be on the safer side updating that value too.

**Review by shreyas098 — APPROVED** [review]

File: lib/classes/manage_location.rb (hunk 3 above)
Comment (swatijadhav) [reply lib/classes/manage_location.rb:9]: yes @patrixr , those records will not be detected.

File: lib/classes/manage_location.rb (hunk 4 above)
Comment (patrixr) [reply lib/classes/manage_location.rb:15]: yup sounds good, I'll create a ticket for removing this column separately

**Review by steveyken — COMMENTED** [review]
Thank you. Good work

**Pull request merged**